backend/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM node:16-alpine
```

Comment:
1) Don't use Alpine unless you really need to
2) Use multi-stage builds, in this case 4:
  * install dependencies (always `npm ci`, not `npm i`!!!)
  * prune dev dependencies
  * build stuff (also based on the first step)
  * copy artifacts from steps 2 and 3 into final image
3) Please explicitly list all of the things you need to include, don't just copy everything, it is bad for documentation